packages/components/src/IDReader/readers/QRReader/Scanner.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+import React, { useEffect, useRef, useState } from 'react'
+import QRScanner from 'qr-scanner'
+import styled from 'styled-components'
+
+interface ScannerProps {
+  onError: (error: 'mount') => void
+  onScan: (data: Record<string, unknown>) => void
+}
+
+const QRReader = styled.div`
+  width: 100%;
+  height: 100%;
+  margin: 0 auto;
+  position: relative;
+
+  @media (max-width: 426px) {
+    width: 100%;
+  }
+`
+const Video = styled.video`
+  width: 100%;
+  height: 100%;
+  object-fit: cover;
+`
+
+const withValidation = (callback: (res: Record<string, unknown>) => void) => {
+  return (res: { data: string }) => {
+    if (Boolean(res.data)) {
+      callback(JSON.parse(res.data))
+    }
+  }
+}
+
+const Scanner = (props: ScannerProps) => {
+  const scanner = useRef<QRScanner>()
+  const videoElement = useRef<HTMLVideoElement>(null)
+  const [qrOn, setQrOn] = useState(true)
```

Comment:
If I read this correctly, qrOn can never be false. When the camera cannot be read, we need to consider the error cases a bit. Let's create a new ticket on that and solve it later.

Reply:
Yes, thanks @naftis  that is true and because of the fact that `qrOn` is never false, 
https://github.com/opencrvs/opencrvs-core/blob/02c283fb204d9baea7161e4806a894425c05b76b/packages/components/src/IDReader/readers/QRReader/Scanner.tsx#L85
this line becomes irrelevant. We might need to determine how to handle that case as the current way is showing an alert which was an interim change and not sure if it should stay like this

Issue created: https://github.com/opencrvs/opencrvs-core/issues/8330